test/e2e/compact_test.go

```diff
@@ -0,0 +1,69 @@
+package e2e
+
+import (
+	"context"
+	"os"
+	"path"
+	"path/filepath"
+	"testing"
+	"time"
+
+	"github.com/cortexproject/cortex/integration/e2e"
+	e2edb "github.com/cortexproject/cortex/integration/e2e/db"
+	"github.com/go-kit/kit/log"
+	"github.com/prometheus/prometheus/pkg/labels"
+	"github.com/prometheus/prometheus/pkg/timestamp"
+	"github.com/thanos-io/thanos/pkg/objstore"
+	"github.com/thanos-io/thanos/pkg/objstore/client"
+	"github.com/thanos-io/thanos/pkg/objstore/s3"
+	"github.com/thanos-io/thanos/pkg/testutil"
+	"github.com/thanos-io/thanos/pkg/testutil/e2eutil"
+	"github.com/thanos-io/thanos/test/e2e/e2ethanos"
+)
+
+func TestCompact(t *testing.T) {
+	t.Parallel()
+	l := log.NewLogfmtLogger(os.Stdout)
+
+	ctx, cancel := context.WithTimeout(context.Background(), 1*time.Minute)
+	defer cancel()
+
+	s, err := e2e.NewScenario("e2e_test_compact")
+	testutil.Ok(t, err)
+	defer s.Close()
+
+	const bucket = "thanos"
+
+	m := e2edb.NewMinio(80, bucket)
+	testutil.Ok(t, s.StartAndWaitReady(m))
+
+	s3Config := s3.Config{
+		Bucket:    bucket,
+		AccessKey: e2edb.MinioAccessKey,
+		SecretKey: e2edb.MinioSecretKey,
+		Endpoint:  m.NetworkHTTPEndpoint(),
+		Insecure:  true,
+	}
+
+	series := []labels.Labels{labels.FromStrings("a", "1", "b", "2")}
```

Comment:
Let's maybe put block creation to separate function, because we need to create tons of them for the tests

Reply:
Potentiall would be nice to have a table test for different tests
* Offline Dedup
* Vertical dedup (let's focus in this)
* Downs
* Compacti
* retention 

etc